integration/spark2/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala

```diff
@@ -105,6 +109,11 @@ object IndexServer extends ServerInterface {
         sparkSession.sparkContext.setLocalProperty("spark.jobGroup.id", request.getTaskGroupId)
         sparkSession.sparkContext
           .setLocalProperty("spark.job.description", request.getTaskGroupDesc)
+        // Fire Generic Event like ACLCheck..etc
```

Comment:
if the event is getting fired only in get splits, one place, then better to make the operationContext as local variable

Reply:
operationContext  will used in all method except ShowCache() Method and those method will be called multiple time so defined global one time object should be ok.